workflow/workflow.go

```diff
@@ -408,79 +414,48 @@ func (l *loopState) notifySteps() { //nolint:gocognit
 				// other copies of this should be attempting to write to the output
 				// data channel. This is required to prevent the goroutine from stalling.
 				l.outputDataChannel <- outputDataType{
-					outputID:   node.Item().OutputID,
+					outputID:   nodeItem.OutputID,
 					outputData: untypedInputData,
 				}
 				// Since this is the only thread accessing the channel, it should be
 				// safe to close it now
 				close(l.outputDataChannel)
 			}
 
-			if err := node.Remove(); err != nil {
+			if err := node.ResolveNode(dgraph.Resolved); err != nil {
 				l.logger.Errorf("BUG: Error occurred while removing workflow output node (%w)", err)
 			}
 		}
 	}
 }
 
 type stateCounters struct {
-	starting              int
-	waitingWithInbound    int
-	waitingWithoutInbound int
-	running               int
-	finished              int
+	starting int
+	waiting  int
+	running  int
+	finished int
 }
 
 func (l *loopState) countStates() stateCounters {
 	counters := struct {
-		starting              int
-		waitingWithInbound    int
-		waitingWithoutInbound int
-		running               int
-		finished              int
+		starting int
+		waiting  int
+		running  int
+		finished int
 	}{
 		0,
 		0,
 		0,
 		0,
-		0,
 	}
```

Comment:
Can't we replace lines 440-450 with:
```go
	counters := stateCounters{}
```
Or, better yet, just declare `counters` as the output parameter:
```go
func (l *loopState) countStates() (counters stateCounters) {
```
and skip the initialization altogether.

Reply:
That seems to work. Done.